core/clustersmngr/factory_caches.go

```diff
@@ -148,3 +149,27 @@ func (un *UsersNamespaces) Clear() {
 func (un UsersNamespaces) cacheKey(user *auth.UserPrincipal, cluster string) uint64 {
 	return ttlcache.StringKey(fmt.Sprintf("%s:%s", user.ID, cluster))
 }
+
+type UsersClients struct {
+	Cache *ttlcache.Cache
+}
+
+func (uc *UsersClients) cacheKey(user *auth.UserPrincipal) uint64 {
+	return ttlcache.StringKey(fmt.Sprintf("%s:%s", user.ID, strings.Join(user.Groups, "/")))
+}
+
+func (uc *UsersClients) Set(user *auth.UserPrincipal, client Client) {
+	uc.Cache.Set(uc.cacheKey(user), client, 5*time.Minute)
```

Comment:
Does the 5 minute timeout mean that if I come back 5 minutes later, things take 7 seconds to load again?. I guess we will find out.

Reply:
very likely, this need to be paired with using pagination, which is implemented in the backend but needs to be used on ui.

Reply:
Is a `cursor` tied to a client?

Reply:
good question, with this caching stuff we'll need to investigate how it plays with our current pagination solution.